cli/cmd/generate.go

```diff
@@ -374,6 +398,43 @@ func runtimeClassNamePatcher(handler string) func(*applycorev1.PodSpecApplyConfi
 	}
 }
 
+func getTmpPaths(paths []string) (string, []string, error) {
+	tmpDir, err := os.MkdirTemp("", "contrast-generate")
+	if err != nil {
+		return "", nil, fmt.Errorf("create temporary directory: %w", err)
+	}
+	var newPaths []string
+	for _, path := range paths {
+		in, err := os.Open(path)
+		if err != nil {
+			return tmpDir, nil, fmt.Errorf("open %s: %w", path, err)
+		}
+		out, err := os.Create(filepath.Join(tmpDir, filepath.Base(path)))
+		if err != nil {
+			return tmpDir, nil, fmt.Errorf("create %s: %w", path, err)
+		}
+		if _, err := io.Copy(out, in); err != nil {
+			return tmpDir, nil, fmt.Errorf("copy %s: %w", path, err)
+		}
+		in.Close()
+		out.Close()
```

Comment:
You should also close these on error - maybe it'd be easier to define a `copyFile` function, or wrap the loop body in a closure.